### PR TITLE
Changes for TypeScript migration of components in gridExplore src/components root directory

### PR DESCRIPTION
**Pull request by FranckLecuyer**

### CODE REVIEW

**Review by anistouri — APPROVED** [review]

File: src/components/filter/explicitNaming/ExplicitNamingFilterEditionDialog.tsx [hunk 1]
```diff
@@ -35,7 +35,7 @@ const formSchema = yup
     .required();
 
 export interface ExplicitNamingFilterEditionDialogProps {
-    id: string;
+    id: string | null;
```
Comment (Tristan-WorkGH) [inline src/components/filter/explicitNaming/ExplicitNamingFilterEditionDialog.tsx:38]: `id` can't be null for `saveExplicitNamingFilter()` no?

File: src/components/filter/criteriaBased/CriteriaBasedFilterEditionDialog.tsx [hunk 2]
```diff
@@ -46,15 +46,15 @@ const formSchema = yup
     .required();
 
 export interface CriteriaBasedFilterEditionDialogProps {
-    id: string;
+    id: string | undefined | null;
```
Comment (Tristan-WorkGH) [inline src/components/filter/criteriaBased/CriteriaBasedFilterEditionDialog.tsx:49]: I don't think that `saveFilter()` support the now _null_ or _undefined_ `id` of `frontToBackTweak()`...

File: src/components/filter/criteriaBased/CriteriaBasedFilterEditionDialog.tsx [hunk 3]
```diff
@@ -46,15 +46,15 @@ const formSchema = yup
...
     name: string;
     titleId: string;
     open: boolean;
     onClose: () => void;
     broadcastChannel: BroadcastChannel;
     getFilterById: (id: string) => Promise<any>;
-    selectionForCopy: SelectionCopy;
-    setSelelectionForCopy: (selection: SelectionCopy) => Dispatch<SetStateAction<SelectionCopy>>;
+    selectionForCopy: any;
+    setSelectionForCopy: (selection: any) => void;
```
Comment (Tristan-WorkGH) [inline src/components/filter/criteriaBased/CriteriaBasedFilterEditionDialog.tsx:57]: Why did you pass this to `any`??  
If there is a problem in gridexplore, either fix the type or add a todo, but don't pass to any just to disable alerts.

**Review by Tristan-WorkGH — CHANGES_REQUESTED** [review]

File: src/components/filter/explicitNaming/ExplicitNamingFilterEditionDialog.tsx (hunk 1 above)
Comment (FranckLecuyer) [reply src/components/filter/explicitNaming/ExplicitNamingFilterEditionDialog.tsx:38]: Same remark : 

currentExplicitNamingFilterId can be null in gridexplore-app, in directory-content.tsx :

<ExplicitNamingFilterEditionDialog
   id={currentExplicitNamingFilterId}

File: src/components/filter/criteriaBased/CriteriaBasedFilterEditionDialog.tsx (hunk 3 above)
Comment (FranckLecuyer) [reply src/components/filter/criteriaBased/CriteriaBasedFilterEditionDialog.tsx:57]: Same has been done in previous typescript migration (ExpertFilterEditionDialog, ExplicitNamingFilterEditionDialog)
And we need to be homogenous between the different filter edition dialogs ...

File: src/components/filter/criteriaBased/CriteriaBasedFilterEditionDialog.tsx (hunk 2 above)
Comment (FranckLecuyer) [reply src/components/filter/criteriaBased/CriteriaBasedFilterEditionDialog.tsx:49]: In practice, id will not be null or undefined when saveFilter will be called ...

File: src/components/filter/criteriaBased/criteriaBasedFilterUtils.ts [hunk 4]
```diff
@@ -155,7 +155,7 @@ function cleanNominalVoltages(formValues: any) {
  *  freeProperties2.{nameA:valuesC}}
  * @author Laurent LAUGARN modified by Florent MILLOT
  */
-export const frontToBackTweak = (id?: string, filter?: any) => {
+export const frontToBackTweak = (id: string | undefined | null, filter?: any) => {
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/components/filter/criteriaBased/criteriaBasedFilterUtils.ts]: Done

**Pull request closed without merging**